Move long-shipping features to the new 'shipping' status

brentfulgham opened the pull request:
#### 094a38c9acd3d93c930091718a563674166b881a
<pre>
Move long-shipping features to the new &apos;shipping&apos; status
<a href="https://bugs.webkit.org/show_bug.cgi?id=250395">https://bugs.webkit.org/show_bug.cgi?id=250395</a>
&lt;rdar://problem/104081353&gt;

Reviewed by Patrick Angle and Elliott Williams.

Following Bug 247926, we expose &apos;stable&apos; features to the Safari &apos;Develop&apos; menu to
support A/B testing by developers so they can isolate the cause of page regression
to specific features.

This has created an overly-cumbersome UI, and surfaced features that cannot be
disabled on the modern web (and which have been shipping for many years).

This patch updates our Feature flag definitions to use &apos;shipping&apos; for long-shipping
features. These will no longer be displayed in the UI, and the feature toggles will
likely be removed in a future update as they are no longer conditional, but are
foundational to the modern web.

This also corrects MediaSourceEnabled and SafeBrowsing to be &apos;embedder&apos;.

* Source/WTF/Scripts/GeneratePreferences.rb: Add &apos;shipping&apos; as a recognized status.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Scripts/GenerateSettings.rb: Add &apos;shipping&apos; as a recognized status.
* Source/WebKit/UIProcess/API/APIFeatureStatus.h
* Source/WebKit/UIProcess/API/Cocoa/_WKFeature.mm
* Source/WebKitLegacy/mac/WebView/WebFeature.h

Canonical link: https://commits.webkit.org/258753@main
</pre>

patrickangle reviewed (approved):
We should probably diff what ends up in "Experimental" features now versus before the unification just to make sure we've accounted for everything (nothing missing and nothing unexpected), but this seems like a logical start.

emw-apple commented on Source/WTF/Scripts/GeneratePreferences.rb:
```diff
@@ -236,7 +236,7 @@ def initialize(preferenceFiles, frontend)
 
   # Corresponds to WebFeatureStatus enum cases. "developer" and up require human-readable names.
-  STATUSES = %w{ embedder unstable internal developer testable preview stable }
+  STATUSES = %w{ embedder unstable internal developer testable preview stable shipping }
```
We should add `WebFeatureStatusShipping` and `API::FeatureStatus::Shipping` cases, too. Right now, it's not a build failure, because `shipping` features aren't part of the experimental/internal-debug-only templates. Once #8345 lands, WebKit will generate code for `shipping` features as part of WebPreferences::features(), and we'll need those enum cases.

brentfulgham replied:
Ah! Gotcha. I just updated with those bits added.

emw-apple replied:
Thank you!

brentfulgham committed:
Move long-shipping features to the new 'shipping' status
https://bugs.webkit.org/show_bug.cgi?id=250395
<rdar://problem/104081353>

Reviewed by Patrick Angle and Elliott Williams.

Following Bug 247926, we expose 'stable' features to the Safari 'Develop' menu to
support A/B testing by developers so they can isolate the cause of page regression
to specific features.

This has created an overly-cumbersome UI, and surfaced features that cannot be
disabled on the modern web (and which have been shipping for many years).

This patch updates our Feature flag definitions to use 'shipping' for long-shipping
features. These will no longer be displayed in the UI, and the feature toggles will
likely be removed in a future update as they are no longer conditional, but are
foundational to the modern web.

This also corrects MediaSourceEnabled and SafeBrowsing to be 'embedder'.

* Source/WTF/Scripts/GeneratePreferences.rb: Add 'shipping' as a recognized status.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/UIProcess/API/APIFeatureStatus.h
* Source/WebKit/UIProcess/API/Cocoa/_WKFeature.mm
* Source/WebKitLegacy/mac/WebView/WebFeature.h

Canonical link: https://commits.webkit.org/258753@main
